src/platform/unix/mod.rs

```diff
@@ -825,7 +880,7 @@ fn recv(fd: c_int, blocking_mode: BlockingMode)
         let bytes_read = try!(cmsg.recv(fd, blocking_mode));
         main_data_buffer.set_len(bytes_read - mem::size_of_val(&total_size));
 
-        let cmsg_fds = cmsg.cmsg_buffer.offset(1) as *const c_int;
+        let cmsg_fds = CMSG_DATA(cmsg.cmsg_buffer) as *const c_int;
```

Comment:
I can submit this as a separate one... There are a few other things I've found that I'll want to fix too